extensions/omni_sql/tests/omni_sql/raw_statements.yml

```diff
@@ -26,3 +26,80 @@ tests:
   - source: select 1
     line: 2
     col: 2 # intentional
+
+- name: with preserve transactions = true but no transaction
+  query: select *
+         from omni_sql.raw_statements(E' Select True;\n select 1', true)
+  results:
+  - source: Select True
+    line: 1
+    col: 2
+  - source: select 1
+    line: 2
+    col: 2
+
+- name: with preserve transactions = true and a commited transaction
+  query: select *
+         from omni_sql.raw_statements(E'begin; Select True;\n select 1; commit;', true)
+  results:
+  - source: begin; Select True; select 1; commit
+    line: 1
+    col: 1
+
+- name: with preserve transactions = true and an end transaction
+  query: select *
+         from omni_sql.raw_statements(E'begin; Select True;\n select 1; end transaction;', true)
+  results:
+  - source: begin; Select True; select 1; end transaction
+    line: 1
+    col: 1
+
+
+- name: with preserve transactions = true and a rollbacked transaction
+  query: select *
+         from omni_sql.raw_statements(E'begin; Select True;\n select 1; rollback;', true)
+  results:
+  - source: begin; Select True; select 1; rollback
+    line: 1
+    col: 1
+
+- name: with preserve transactions = true and an unfinished transaction
```

Comment:
I am trying to figure out here if this is how it should work.

If I take this expression and it send it.. it's an unfinished transaction hanging there.

If we ungroup it, then the net effect is the same.

Should we actually error out here? And then, perhaps, error out if we encounter another begin in an open transaction?

Reply:
Totally special-casing here, but what if we had another optional parameter saying what to do with this situation: auto-rollback, leave as is, don't group?

Reply:
I would avoid additional parameters until there is a concrete use case. The current implementation pushes the problem down the line. Raising an error would potentially save the user some headache.

Reply:
I thought a bit more about this, and could not see any interesting use case for the unfinished transaction. So I decided to send a commit going down the simpler and safer approach and just error on the unfinished case.

Reply:
I left this last change in a separate commit in case you rather drop it and keep the previous behavior.